controller/collaborators.go

```diff
@@ -142,12 +157,59 @@ func (c *CollaboratorsController) AddMany(ctx *app.AddManyCollaboratorsContext)
 			return jsonapi.JSONErrorResponse(ctx, err)
 		}
 	}
+
+	// Now assign contributor role to the collaborators
+	err = c.addContributors(ctx, currentUser.ID, ctx.Payload.Data, ctx.SpaceID.String())
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err":      err,
+			"space_id": ctx.SpaceID,
+		}, "unable to add contributors to space resource: resource not found; that's OK for old spaces")
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
 	return ctx.OK([]byte{})
 }
 
+func (c *CollaboratorsController) addContributors(ctx context.Context, byIdentityID uuid.UUID, contributors []*app.UpdateUserID, spaceID string) error {
+	err := c.app.ResourceRepository().CheckExists(ctx, spaceID)
+	if err != nil {
+		if notFound, _ := autherrors.IsNotFoundError(err); notFound {
+			log.Warn(ctx, map[string]interface{}{
+				"err":      err,
+				"space_id": spaceID,
+			}, "unable to add contributors to space resource: resource not found; that's OK for old spaces")
+			// Just log the warning. Old spaces doesn't have any registered resources.
+			return nil
+		}
+		return err
+	}
+
+	err = c.app.PermissionService().RequireScope(ctx, byIdentityID, spaceID, authorization.ManageRoleAssignmentsInSpaceScope)
+	if err != nil {
+		return err
+	}
+
+	res := resource.Resource{ResourceType: resourcetype.ResourceType{Name: authorization.ResourceTypeSpace}, ResourceID: spaceID}
+	for _, contributor := range contributors {
+		identityID, err := uuid.FromString(contributor.ID)
+		if err != nil {
+			return autherrors.NewBadParameterError("ids", contributor.ID).Expected("uuid")
+		}
+
+		// Have to use ForceAssign() because Assign() requires assignees to already have any role in the space
+		err = c.app.RoleManagementService().ForceAssign(ctx, identityID, authorization.SpaceContributorRole, res)
```

Comment:
There is no transaction wrapping this code?

Reply:
Each forceAssign will create a new transaction. Not optimal but IMO OK for the collaborators API because we are deprecating it anyway.